File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        
        // participant = 참가자, completion = 완주자
        // 참가자를 map에 삽입
        // 완주자를 map에서 하나씩 제거
        // map에 남은 사람은 미완주자
        
        Map<String,Integer> map = new HashMap();
        for(int i = 0; i < participant.length; i++){
            
            String participantMan = participant[i];
            
            if(map.containsKey(participantMan)){
                map.replace(participantMan,map.get(participantMan)+1);
            }else{
                map.put(participantMan,1);
            }
        }
        
        StringBuffer sb = new StringBuffer();
        for(int i = 0; i < completion.length; i++){
            
            String completeMan = completion[i];
            
            if(map.containsKey(completeMan)){
                
                map.replace(completeMan,map.get(completeMan)-1);
            
                if(map.get(completeMan)==0){
                    map.remove(completeMan,map.get(completeMan));
                }
            }
        }
        
        for(String unCompleteMan : map.keySet()){
            sb.append(unCompleteMan);
        }
        
        String answer = sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    
    var map = new Map();
    
    // 참가자 입력
    for(var i = 0; i < participant.length; i++){
        if(map.has(participant[i])){
            map.set(participant[i],map.get(participant[i])+1);            
        }else{
            map.set(participant[i],1);
        }
    }
    
    // 완주자 제거
    for(var i = 0; i < completion.length; i++){
        map.set(completion[i],map.get(completion[i])-1);
    }
    
    
    for(var name of map.keys()){
        if(map.get(name)===1){
            answer = name;
            break;
        }
    }
        
    
    return answer;
}